src/contexts/SendContext.tsx

```diff
@@ -0,0 +1,210 @@
+import { node } from '@alephium/web3'
+import { createContext, useCallback, useContext, useState } from 'react'
+import Toast from 'react-native-root-toast'
+
+import { buildSweepTransactions, buildUnsignedTransactions, signAndSendTransaction } from '~/api/transactions'
+import ConfirmWithAuthModal from '~/components/ConfirmWithAuthModal'
+import ConsolidationModal from '~/components/ConsolidationModal'
+import { useAppDispatch, useAppSelector } from '~/hooks/redux'
+import { selectAddressByHash, transactionSent } from '~/store/addressesSlice'
+import { AddressHash } from '~/types/addresses'
+import { getTransactionAssetAmounts } from '~/utils/transactions'
+
+type UnsignedTxData = {
+  unsignedTxs: {
+    txId: node.BuildTransactionResult['txId'] | node.SweepAddressTransaction['txId']
+    unsignedTx: node.BuildTransactionResult['unsignedTx'] | node.SweepAddressTransaction['unsignedTx']
+  }[]
+  fees: bigint
+}
+
+type BuildTransactionCallbacks = {
+  onBuildSuccess: () => void
+  onConsolidationSuccess: () => void
+}
+
+interface SendContextProps {
+  toAddress?: AddressHash
+  setToAddress: (toAddress: AddressHash) => void
+  fromAddress?: AddressHash
+  setFromAddress: (toAddress: AddressHash) => void
+  assetAmounts: AssetAmount[]
+  setAssetAmount: (assetId: string, amount?: bigint) => void
+  fees: bigint
+  buildTransaction: (callbacks: BuildTransactionCallbacks) => Promise<void>
+  sendTransaction: (onSendSuccess: () => void) => Promise<void>
+}
+
+const initialValues: SendContextProps = {
+  toAddress: undefined,
+  setToAddress: () => null,
+  fromAddress: undefined,
+  setFromAddress: () => null,
+  assetAmounts: [],
+  setAssetAmount: () => null,
+  fees: BigInt(0),
+  buildTransaction: () => Promise.resolve(undefined),
+  sendTransaction: () => Promise.resolve(undefined)
+}
+
+const SendContext = createContext(initialValues)
+
+export const SendContextProvider: FC = ({ children }) => {
+  const requiresAuth = useAppSelector((s) => s.settings.requireAuth)
+  const dispatch = useAppDispatch()
+
+  const [toAddress, setToAddress] = useState<SendContextProps['toAddress']>(initialValues.toAddress)
+  const [fromAddress, setFromAddress] = useState<SendContextProps['fromAddress']>(initialValues.fromAddress)
+  const [assetAmounts, setAssetAmounts] = useState<SendContextProps['assetAmounts']>(initialValues.assetAmounts)
+  const [unsignedTxData, setUnsignedTxData] = useState<UnsignedTxData>({ unsignedTxs: [], fees: initialValues.fees })
+
+  const [consolidationRequired, setConsolidationRequired] = useState(false)
+  const [isConsolidateModalVisible, setIsConsolidateModalVisible] = useState(false)
+  const [isAuthenticationModalVisible, setIsAuthenticationModalVisible] = useState(false)
+  const [onSendSuccessCallback, setOnSendSuccessCallback] = useState<() => void>(() => () => null)
+
+  const address = useAppSelector((s) => selectAddressByHash(s, fromAddress ?? ''))
+
+  const setAssetAmount = (assetId: string, amount?: bigint) => {
```

Comment:
Idea: what about using a `Map<AssetId, bigint>` for this? This would make the get / set processes that much easier! It would also remove the need for storing `set` functions in the global context.

About using Maps with React: 
- https://usehooks-ts.com/react-hook/use-map
- https://azimi.io/es6-map-with-react-usestate-9175cd7b409b

Reply:
I know what you're going to say, that we're used to specific `set` functions when using `setState`, so it makes sense to build the context in a similar way. But I find this idea interesting anyway: we never use Maps, however we often have cases where we manipulate arrays of key/values...

Reply:
Just to be clear, you recommend to instead of using `assetAmounts: AssetAmount[]` array in the `SendContextProps` to use a `Map<AssetId, bigint>`?

Reply:
> It would also remove the need for storing set functions in the global context.

I don't understand this part!

---

Thanks for the links!